### PR TITLE
Address didTransition deprecation

### PR DESCRIPTION
**Pull request by locks**
Things I have tried:

- `this.on('routeDidChange'` in EmberRouter's `init` - handler not called
- Inject the router service in EmberRouter and add the handler on init - max call stack reached
- Inject the router service in application route and add the handler on init - handler is not called

### CODE REVIEW

**Comment by acorncom** [comment]
@locks looks like you have failing tests

File: app/router.js [hunk 1]
```diff
@@ -6,16 +6,19 @@ import { get } from '@ember/object';
...
   metrics: service(),
   fastboot: service(),
 
-  didTransition() {
+  init() {
     this._super(...arguments);
-    this.navbar.closePopupMenu();
-    this._trackPage();
+
+    this.on('routeDidChange', () => {
```
Comment (rwjblue) [inline app/router.js]:
```suggestion
    this.router.on('routeDidChange', () => {
```

File: app/router.js [hunk 2]
```diff
@@ -6,16 +6,19 @@ import { get } from '@ember/object';
 import { scheduleOnce } from '@ember/runloop';
 
 const Router = EmberRouter.extend({
-  navbar: service(),
   location: config.locationType,
   rootURL: config.rootURL,
+  navbar: service(),
   metrics: service(),
   fastboot: service(),
```
Comment (rwjblue) [inline app/router.js]:
```suggestion
  fastboot: service(),
  router: service(),
```

Comment (locks) [reply app/router.js]: I've tried this and I get an error:

```
Uncaught RangeError: Maximum call stack size exceeded at http://localhost:7357/assets/vendor.js, line 23173
```

Comment (rwjblue) [reply app/router.js]: Tweaked the suggestion...

Comment (locks) [reply app/router.js]: I can't the invocation right, and I don't know what I'm doing wrong 🤔

Comment (acorncom) [reply app/router.js]: @locks would upgrading beyond the Ember 3.5.x that we’re on here help?

**Comment by mansona** [comment]
@locks @rwjblue is this way of doing router `routeDidChange` stuff in the **application** route recommended? I want to make sure that we're doing the most recommended thing here and make sure that we bring this documentation upstream to the right places (deprecations documentation or ember-metrics)

**Comment by locks** [comment]
@mansona yes, and I've opened an issue for updating the deprecation guide. There is a dependency loop between `EmberRouter` and `RouterService` that prevents us from doing it in `EmberRouter`.

**Comment by mansona** [comment]
@locks thanks a million for following up on this 🎉

**Pull request merged**